renderer/components/nettests/middleboxes/HttpHeaderFieldManipulation.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { FormattedMessage } from 'react-intl'
+import { Flex, Box } from 'ooni-components'
+
+import FormattedMarkdownMessage from '../../FormattedMarkdownMessage'
+
+const HttpHeaderFieldManipulation = ({isAnomaly, render}) => {
+
+  const HHFMDetails = () => (
+    <Box>
+      <Flex my={3} flexDirection='column'>
+        {isAnomaly ? (
+          <Box>
+            <FormattedMarkdownMessage
+              id='TestResults.Details.Middleboxes.HTTPHeaderFieldManipulation.Found.Content.Paragraph'
+            />
+          </Box>
+        ): (
+          <Box>
+            <FormattedMessage
+              id='TestResults.Details.Middleboxes.HTTPHeaderFieldManipulation.NotFound.Content.Paragraph'
+            />
+          </Box>
+        )}
+      </Flex>
+    </Box>
+  )
+
+  const heroTitle = isAnomaly ? (
+    <FormattedMessage id='TestResults.Details.Middleboxes.HTTPHeaderFieldManipulation.Found.Hero.Title' />
+  ) : (
+    <FormattedMessage id='TestResults.Details.Middleboxes.HTTPHeaderFieldManipulation.NotFound.Hero.Title' />
+  )
+
+  return (
+    <div>
+      {render({
+        heroTitle: heroTitle,
+        details: <HHFMDetails />
+      })}
+    </div>
+  )
+}
+
+HttpHeaderFieldManipulation.propTypes = {
+  measurement: PropTypes.object,
+  isAnomaly: PropTypes.bool,
+  render: PropTypes.func
+}
```

Comment:
This is a rather basic question, which however it's probably worth asking. `HttpHeaderFieldManipulation` is a function that maps `isAnomaly, render` to a rendered div. Then you are adding a `.propTypes` field to the function, which is fine, because a function is a normal JavaScript object, except that it's callable.

Upon closer inspection, this seems the declaration of a "prototype". You are telling React (?), that `measurement` is an object, `isAnomaly` is a boolean, and `render` is a function.

Why is `measurement` required? Where is it used?